linkerd/cache/src/lib.rs

```diff
@@ -1,113 +1,258 @@
 #![deny(warnings, rust_2018_idioms)]
 
-use linkerd2_stack::NewService;
+use linkerd2_stack::{layer, NewService};
 use parking_lot::RwLock;
 use std::{
     collections::{hash_map::Entry, HashMap},
     hash::Hash,
     sync::{Arc, Weak},
+    task::{Context, Poll},
 };
-use tracing::{debug, trace};
-
-pub mod layer;
-
-pub use self::layer::CacheLayer;
+use tokio::time;
+use tracing::{debug, debug_span, trace};
+use tracing_futures::Instrument;
 
 #[derive(Clone)]
-pub struct Cache<T, N>
+pub struct Cache<T, N, S>
```

Comment:
is there any reason for adding a separate type param here rather than keeping it as `N::Service`? i notice that all the impls are for `Cache<T, N, N::Service>`...

Reply:
Probably motivated by an intermediate change. reverted.